x-pack/plugins/alerting/common/alert_schema/field_maps/alert_field_map.ts

```diff
@@ -27,92 +30,94 @@ import {
   ALERT_UUID,
   ALERT_WORKFLOW_STATUS,
   SPACE_IDS,
+  TIMESTAMP,
   VERSION,
 } from '@kbn/rule-data-utils';
 
 export const alertFieldMap = {
-  [ALERT_RULE_PARAMETERS]: {
-    type: 'object',
-    enabled: false,
+  [ALERT_ACTION_GROUP]: {
+    type: 'keyword',
+    array: false,
     required: false,
   },
-  [ALERT_RULE_TYPE_ID]: {
+  [ALERT_CASE_IDS]: {
     type: 'keyword',
+    array: true,
+    required: false,
+  },
+  [ALERT_DURATION]: {
+    type: 'long',
     array: false,
-    required: true,
+    required: false,
   },
-  [ALERT_RULE_CONSUMER]: {
-    type: 'keyword',
+  [ALERT_END]: {
+    type: 'date',
     array: false,
-    required: true,
+    required: false,
   },
-  [ALERT_RULE_PRODUCER]: {
-    type: 'keyword',
+  [ALERT_FLAPPING]: {
+    type: 'boolean',
     array: false,
-    required: true,
+    required: false,
   },
-  [SPACE_IDS]: {
-    type: 'keyword',
+  [ALERT_FLAPPING_HISTORY]: {
+    type: 'boolean',
     array: true,
-    required: true,
-  },
-  [ALERT_UUID]: {
-    type: 'keyword',
-    array: false,
-    required: true,
+    required: false,
   },
-  [ALERT_ID]: {
+  [ALERT_INSTANCE_ID]: {
     type: 'keyword',
     array: false,
     required: true,
   },
-  [ALERT_START]: {
+  // this is not in the technical field mapping ??
+  [ALERT_LAST_DETECTED]: {
     type: 'date',
-    array: false,
     required: false,
-  },
-  [ALERT_TIME_RANGE]: {
-    type: 'date_range',
-    format: 'epoch_millis||strict_date_optional_time',
     array: false,
-    required: false,
   },
-  [ALERT_END]: {
-    type: 'date',
+  [ALERT_REASON]: {
+    type: 'keyword',
     array: false,
     required: false,
   },
-  [ALERT_DURATION]: {
-    type: 'long',
+  [ALERT_RULE_CATEGORY]: {
+    type: 'keyword',
     array: false,
-    required: false,
+    required: true,
   },
-  [ALERT_STATUS]: {
+  [ALERT_RULE_CONSUMER]: {
     type: 'keyword',
     array: false,
     required: true,
   },
-  [VERSION]: {
-    type: 'version',
+  [ALERT_RULE_EXECUTION_UUID]: {
+    type: 'keyword',
     array: false,
     required: false,
   },
-  [ALERT_WORKFLOW_STATUS]: {
+  [ALERT_RULE_NAME]: {
     type: 'keyword',
     array: false,
-    required: false,
+    required: true,
   },
-  [ALERT_ACTION_GROUP]: {
-    type: 'keyword',
+  [ALERT_RULE_PARAMETERS]: {
```

Comment:
We have to keep the `flattened` mapping for the rule parameters to maintain backwards compatibility